targets/TARGET_STM/pwmout_api.c

```diff
@@ -180,92 +211,54 @@ void pwmout_period_us(pwmout_t* obj, int us)
-#endif
-#if defined(TIM4_BASE)
-        case PWM_4:
-#endif
-#if defined(TIM5_BASE)
-        case PWM_5:
-#endif
-#if defined(TIM12_BASE)
-        case PWM_12:
-#endif
-#if defined(TIM13_BASE)
-        case PWM_13:
-#endif
-#if defined(TIM14_BASE)
-        case PWM_14:
-#endif
-            PclkFreq = HAL_RCC_GetPCLK1Freq();
-            APBxCLKDivider = RCC_ClkInitStruct.APB1CLKDivider;
-            break;
-        
-        // APB2 clock
-#if defined(TIM1_BASE)
-        case PWM_1:
-#endif
-#if defined(TIM8_BASE)
-        case PWM_8:
-#endif
-#if defined(TIM9_BASE)
-        case PWM_9:
-#endif
-#if defined(TIM10_BASE)
-        case PWM_10:
-#endif
-#if defined(TIM11_BASE)
-        case PWM_11:
-#endif
-            PclkFreq = HAL_RCC_GetPCLK2Freq();
-            APBxCLKDivider = RCC_ClkInitStruct.APB2CLKDivider;
-            break;
-        default:
-            return;
+    /*  Parse the pwm / apb mapping table to find the right entry */
+    while(pwm_apb_map_table[i].pwm != obj->pwm) {
+        i++;
     }
 
-    /* To make it simple, we use to possible prescaler values which lead to:
-     * pwm unit = 1us, period/pulse can be from 1us to 65535us
-     * or
-     * pwm unit = 500us, period/pulse can be from 500us to ~32.76sec
-     * Be careful that all the channels of a PWM shares the same prescaler
-     */
-    if (us >  0xFFFF) {
-        obj->prescaler = 500;
+    if(pwm_apb_map_table[i].pwm == 0)
+        error("Unknown PWM instance");
+
+    if(pwm_apb_map_table[i].pwmoutApb == PWMOUT_ON_APB1) {
+        PclkFreq = HAL_RCC_GetPCLK1Freq();
+        APBxCLKDivider = RCC_ClkInitStruct.APB1CLKDivider;
     } else {
-        obj->prescaler = 1;
+#if !defined(PWMOUT_APB2_NOT_SUPPORTED)
+        PclkFreq = HAL_RCC_GetPCLK2Freq();
+        APBxCLKDivider = RCC_ClkInitStruct.APB2CLKDivider;
+#endif
     }
 
+
+    /* By default use, 1us as SW pre-scaler */
+    obj->prescaler = 1;
     // TIMxCLK = PCLKx when the APB prescaler = 1 else TIMxCLK = 2 * PCLKx
     if (APBxCLKDivider == RCC_HCLK_DIV1)
-      TimHandle.Init.Prescaler   = (uint16_t)(((PclkFreq) / 1000000) * obj->prescaler) - 1; // 1 us tick
+        TimHandle.Init.Prescaler   = (((PclkFreq) / 1000000)) - 1; // 1 us tick
     else
-      TimHandle.Init.Prescaler   = (uint16_t)(((PclkFreq * 2) / 1000000) * obj->prescaler) - 1; // 1 us tick
-
-    if (TimHandle.Init.Prescaler > 0xFFFF)
-        error("PWM: out of range prescaler");
-
-    TimHandle.Init.Period        = (us - 1) / obj->prescaler;
-    if (TimHandle.Init.Period > 0xFFFF)
-        error("PWM: out of range period");
+        TimHandle.Init.Prescaler   = (((PclkFreq * 2) / 1000000)) - 1; // 1 us tick
+    TimHandle.Init.Period        = (us - 1);
+
+    /*  In case period or pre-scalers are out of range, loop-in to get valid values */
+    while ((TimHandle.Init.Period > 0xFFFF) || (TimHandle.Init.Period > 0xFFFF)) {
+        obj->prescaler = obj->prescaler * 2;
+        if (APBxCLKDivider == RCC_HCLK_DIV1)
```

Comment:
@0xc0170 I had a quick look but probably missed the point - what's wrong with current formatting ?